recommend/admissionControllerPolicy.go

```diff
@@ -0,0 +1,84 @@
+package recommend
+
+import (
+	"context"
+	"errors"
+	"github.com/accuknox/auto-policy-discovery/src/libs"
+	"github.com/accuknox/auto-policy-discovery/src/protobuf/v1/worker"
+	"github.com/clarketm/json"
+	"github.com/fatih/color"
+	"github.com/kubearmor/kubearmor-client/k8s"
+	"github.com/kubearmor/kubearmor-client/utils"
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func recommendAdmissionControllerPolicies(img ImageInfo, c *k8s.Client) error {
+	labels := libs.LabelMapToString(img.Labels)
+	gRPC := ""
+	targetSvc := "discovery-engine"
+	var port int64 = 9089
+	mtchLabels := map[string]string{"app": "discovery-engine"}
+	if val, ok := os.LookupEnv("DISCOVERY_SERVICE"); ok {
+		gRPC = val
+	} else {
+		pf, err := utils.InitiatePortForward(c, port, port, mtchLabels, targetSvc)
+		if err != nil {
+			return err
+		}
+		gRPC = "localhost:" + strconv.FormatInt(pf.LocalPort, 10)
+	}
+	// create a client
+	conn, err := grpc.Dial(gRPC, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
Can you please put this into a function and refactor the code base?

Reply:
Done.